### PR TITLE
Fix aptos example

### PR DESCRIPTION
**Pull request by npty**
# Description

- Fix Aptos examples
- Fix `execute` script to handle Aptos example separately.
- Add tests for Aptos examples
- Spin up Aptos local testnet in GitHub Actions for running tests.

### CODE REVIEW

File: examples/aptos/call-contract/README.md [hunk 1]
```diff
@@ -1,37 +1,38 @@
 # Call Contract
 
-Relay a message from source-chain to destination-chain.
+Relay a message bi-directional between Aptos and an EVM chain.
 
 Deploy:
 
 ```bash
-npm run deploy aptos/call-contract [local|testnet]
+npm run deploy aptos/call-contract local
```
Comment (StephenFluin) [inline examples/aptos/call-contract/README.md:8]: Causing error

```
$ npm run deploy aptos/call-contract local

> axelar-examples@1.0.0 deploy
> node scripts/runDeploy "aptos/call-contract" "local"

Deploying HelloWorld for Aptos.
/home/stephen/workspace/axelar/axelar-examples/node_modules/aptos/dist/index.js:5694
        throw new ApiError2(
              ^

ApiError2: {"message":"Account not found by Address(0x8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0) and Ledger version(404)","error_code":"account_not_found","vm_error_code":null}
    at AptosNetwork.wrapper (/home/stephen/workspace/axelar/axelar-examples/node_modules/aptos/dist/index.js:5694:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 0)
    at async AptosNetwork.generateRawTransaction (/home/stephen/workspace/axelar/axelar-examples/node_modules/aptos/dist/index.js:5478:91)
    at async AptosNetwork.generateSignSubmitTransaction (/home/stephen/workspace/axelar/axelar-examples/node_modules/aptos/dist/index.js:5500:28) {
  status: 404,
  errorCode: 'account_not_found',
  vmErrorCode: null
}
```

File: examples/aptos/README.md [hunk 2]
```diff
@@ -2,27 +2,31 @@
...
 ```bash
 aptos node run-local-testnet --with-faucet
 ```
 
-3. Run the `evm` local network
+4. Run the local server
 
 ```bash
-npm scripts/start
+npm run start
```
Comment (StephenFluin) [inline examples/aptos/README.md:27]: Error
```
$ npm run start

> axelar-examples@1.0.0 start
> node scripts/runStart

Error: ENOENT: no such file or directory, open 'node_modules/@axelar-network/axelar-cgp-aptos/aptos/modules/axelar/build/AxelarFramework/package-metadata.bcs'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at AptosNetwork.<anonymous> (/home/stephen/workspace/axelar/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:64:50)
    at Generator.next (<anonymous>)
    at /home/stephen/workspace/axelar/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/stephen/workspace/axelar/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:4:12)
    at AptosNetwork.deploy (/home/stephen/workspace/axelar/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:63:16)
    at AptosNetwork.deployAxelarFrameworkModules (/home/stephen/workspace/axelar/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:96:21)
    at /home/stephen/workspace/axelar/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/aptosNetworkUtils.js:27:46 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'node_modules/@axelar-network/axelar-cgp-aptos/aptos/modules/axelar/build/AxelarFramework/package-metadata.bcs'
}
Skip Aptos initlization, rerun this after starting an aptos node for proper support.

```

Comment (StephenFluin) [inline examples/aptos/README.md:27]: Error

```
$ npm run start

> axelar-examples@1.0.0 start
> node scripts/runStart

AxiosError: connect ECONNREFUSED 127.0.0.1:8081
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16) {
  port: 8081,
  address: '127.0.0.1',
  syscall: 'connect',
  code: 'ECONNREFUSED',
  errno: -111,
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: { FormData: [Function] },
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/x-www-form-urlencoded',
      'User-Agent': 'axios/0.27.2'
    },
    url: 'http://localhost:8081/mint?address=8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0&amount=10000000000',
    method: 'post',
    withCredentials: false,
    cancelToken: CancelToken { promise: [Promise], _listeners: [] },
    data: undefined
  },
  request: <ref *1> Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: false,
      closeEmitted: false,
      [Symbol(kOnFinished)]: []
    },
    _events: [Object: null prototype] {
      response: [Function: handleResponse],
      error: [Function: handleRequestError],
      socket: [Function: handleRequestSocket]
    },
    _eventsCount: 3,
    _maxListeners: undefined,
    _options: {
      maxRedirects: 21,
      maxBodyLength: 10485760,
      protocol: 'http:',
      path: '/mint?address=8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0&amount=10000000000',
      method: 'POST',
      headers: [Object],
      agent: undefined,
      agents: [Object],
      auth: undefined,
      hostname: 'localhost',
      port: '8081',
      nativeProtocols: [Object],
      pathname: '/mint',
      search: '?address=8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0&amount=10000000000'
    },
    _ended: true,
    _ending: true,
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 0,
    _requestBodyBuffers: [],
    _onNativeResponse: [Function (anonymous)],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      _closed: false,
      socket: [Socket],
      _header: 'POST /mint?address=8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0&amount=10000000000 HTTP/1.1\r\n' +
        'Accept: application/json\r\n' +
        'Content-Type: application/x-www-form-urlencoded\r\n' +
        'User-Agent: axios/0.27.2\r\n' +
        'Host: localhost:8081\r\n' +
        'Connection: close\r\n' +
        'Content-Length: 0\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/mint?address=8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0&amount=10000000000',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'localhost',
      protocol: 'http:',
      _redirectable: [Circular *1],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    _currentUrl: 'http://localhost:8081/mint?address=8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0&amount=10000000000',
    [Symbol(kCapture)]: false
  }
}
Skip Aptos initlization, rerun this after starting an aptos node for proper support.
```

**Review by StephenFluin — CHANGES_REQUESTED** [review]

File: examples/aptos/call-contract/README.md (hunk 1 above)
Comment (npty) [reply examples/aptos/call-contract/README.md:8]: @StephenFluin could you try adding this to `.env` file

```
APTOS_ADDRESS=0x8ac1b8ff9583ac8e661c7f0ee462698c57bb7fc454f587e3fa25a57f9406acc0
```

Comment (StephenFluin) [reply examples/aptos/call-contract/README.md:8]: What if they have an old copy of axelar-examples. Could we have the setup script fix this, or detect it and throw a better error?

File: examples/aptos/call-contract/README.md [hunk 3]
```diff
@@ -1,37 +1,38 @@
 # Call Contract
 
-Relay a message from source-chain to destination-chain.
+Relay a message bi-directional between Aptos and an EVM chain.
```
Comment (StephenFluin) [inline examples/aptos/call-contract/README.md:3]: Now getting
```
$ npm run deploy aptos/call-contract local
 execute aptos/call-contract local "Avalanche" 'Hello World'
> axelar-examples@1.0.0 deploy
> node scripts/runDeploy aptos/call-contract local

Deploying HelloWorld for Aptos.
Error: Move abort in 0x1::code: EPACKAGE_DEP_MISSING(0x60005): Dependency could not be resolved to any published package.
Deployed HelloWorld for Aptos.
Deploying HelloWorld for Moonbeam.
Deploying HelloWorld for Avalanche.
Deploying HelloWorld for Fantom.
Deploying HelloWorld for Ethereum.
Deploying HelloWorld for Polygon.
Deployed HelloWorld for Moonbeam at 0xe491FA32130F95f1FF94292e11851CfaC28E4584.
Deployed HelloWorld for Avalanche at 0xe491FA32130F95f1FF94292e11851CfaC28E4584.
Deployed HelloWorld for Ethereum at 0xe491FA32130F95f1FF94292e11851CfaC28E4584.
Deployed HelloWorld for Polygon at 0xe491FA32130F95f1FF94292e11851CfaC28E4584.
Deployed HelloWorld for Fantom at 0xe491FA32130F95f1FF94292e11851CfaC28E4584.
```

**Review by StephenFluin — CHANGES_REQUESTED** [review]
Aptos servers coming up right, now just an error in aptos call-contract.

**Comment by StephenFluin** [comment]
@npty It looks like you need to add .aptos/ to `.gitignore` as well.

File: examples/aptos/README.md (hunk 2 above)
Comment (StephenFluin) [reply examples/aptos/README.md:27]: I verified I ran `npm ci` and waited for the faucet to come up, but I'm still getting this and not able to proceed.

```
$ npm run start

> axelar-examples@1.0.0 start
> node scripts/runStart

Error: ENOENT: no such file or directory, open 'node_modules/@axelar-network/axelar-cgp-aptos/aptos/modules/axelar/build/AxelarFramework/package-metadata.bcs'
    at Object.openSync (node:fs:590:3)
    at Object.readFileSync (node:fs:458:35)
    at AptosNetwork.<anonymous> (/home/stephen/workspace/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:64:50)
    at Generator.next (<anonymous>)
    at /home/stephen/workspace/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/stephen/workspace/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:4:12)
    at AptosNetwork.deploy (/home/stephen/workspace/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:63:16)
    at AptosNetwork.deployAxelarFrameworkModules (/home/stephen/workspace/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/AptosNetwork.js:96:21)
    at /home/stephen/workspace/axelar-examples/node_modules/@axelar-network/axelar-local-dev/dist/aptos/aptosNetworkUtils.js:27:46 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'node_modules/@axelar-network/axelar-cgp-aptos/aptos/modules/axelar/build/AxelarFramework/package-metadata.bcs'
}
Skip Aptos initlization, rerun this after starting an aptos node for proper support.
Creating Moonbeam with a chainId of 2500...
Deploying the ConstAddressDeployer for Moonbeam...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the ConstAddressDeployer for Moonbeam...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the Axelar Gateway for Moonbeam...
Deployed at 0x9E404e6ff4F2a15C99365Bd6615fCE3FB9E9Cb76
Deploying the Axelar Gas Receiver for Moonbeam...
Deployed at 0x783ce2eF32Aa74B41c8EbbbeC6F632b6Da00C1e9
Deploying the Express Service Contract for Moonbeam...
Deployed ExpressService at 0x5A8b2B50a73E83aAfE73c636ed2D11db9c0e8203
Deploying Axelar Wrapped aUSDC for Moonbeam...
Deployed at 0x5d8C7D94B16a1Ba4865acC6bc039e2877925dC4F
Creating Avalanche with a chainId of 2501...
Deploying the ConstAddressDeployer for Avalanche...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the ConstAddressDeployer for Avalanche...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the Axelar Gateway for Avalanche...
Deployed at 0x54C156f087EBA12a869E0CBb49548c3F55E80edC
Deploying the Axelar Gas Receiver for Avalanche...
Deployed at 0xD3cB555CB530C3374B122242b15d2943445c31d6
Deploying the Express Service Contract for Avalanche...
Deployed ExpressService at 0xC89c53D6C7031DA523387f72588380900955533D
Deploying Axelar Wrapped aUSDC for Avalanche...
Deployed at 0x19a13F9C3820B148e68E5aEb67cD1BE20c085CD0
Creating Fantom with a chainId of 2502...
Deploying the ConstAddressDeployer for Fantom...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the ConstAddressDeployer for Fantom...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the Axelar Gateway for Fantom...
Deployed at 0xcb189eB52ca573eFD633d07A3B5357e4d989D743
Deploying the Axelar Gas Receiver for Fantom...
Deployed at 0x85Fa9202C6Be69e9889CC0247Af72ABc70DbF542
Deploying the Express Service Contract for Fantom...
Deployed ExpressService at 0xfFA0EF8701960F8292d3aE5f6AC22C7fA70Da0d6
Deploying Axelar Wrapped aUSDC for Fantom...
Deployed at 0x33251f191aa60F7AA30A5323Eb301557c95f1867
Creating Ethereum with a chainId of 2503...
Deploying the ConstAddressDeployer for Ethereum...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the ConstAddressDeployer for Ethereum...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the Axelar Gateway for Ethereum...
Deployed at 0x013459EC3E8Aeced878C5C4bFfe126A366cd19E9
Deploying the Axelar Gas Receiver for Ethereum...
Deployed at 0x28f8B50E1Be6152da35e923602a2641491E71Ed8
Deploying the Express Service Contract for Ethereum...
Deployed ExpressService at 0xc918b5BA9127677fDbc39C5e3005047F6565dce6
Deploying Axelar Wrapped aUSDC for Ethereum...
Deployed at 0x38Aa1Cb12E5263eC0c6e9febC25B01116D346CD4
Creating Polygon with a chainId of 2504...
Deploying the ConstAddressDeployer for Polygon...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the ConstAddressDeployer for Polygon...
Deployed at 0x69aeB7Dc4f2A86873Dae8D753DE89326Cf90a77a
Deploying the Axelar Gateway for Polygon...
Deployed at 0xc7B788E88BAaB770A6d4936cdcCcd5250E1bbAd8
Deploying the Axelar Gas Receiver for Polygon...
Deployed at 0xC573c722e21eD7fadD38A8f189818433e01Ae466
Deploying the Express Service Contract for Polygon...
Deployed ExpressService at 0x194598519587185673fec0E2C3e9F9Ed94bF2451
Deploying Axelar Wrapped aUSDC for Polygon...
Deployed at 0xC1a50e3611C2c500562803945C0a746c9cFb1939
Serving 5 networks on port 8500
```

**Review by StephenFluin — CHANGES_REQUESTED** [review]

File: examples/aptos/call-contract/README.md (hunk 1 above)
Comment (npty) [reply examples/aptos/call-contract/README.md:8]: good idea. i've added a condition to check if all necessary variables are set in the `build.sh` file.
https://github.com/axelarnetwork/axelar-examples/pull/98/commits/b832653556c91df7b171859331674ca88b763365

**Comment by npty** [comment]
> I'm still getting errors. :(. New ones though! :)
> 
> image
> 
> ```
> $ npm run build-aptos
> 
> > axelar-examples@1.0.0 build-aptos
> > bash ./build-aptos.sh
> 
> Compiling, may take a little while to download git dependencies...
> FETCHING GIT DEPENDENCY https://github.com/axelarnetwork/axelar-cgp-aptos.git
> UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
> INCLUDING DEPENDENCY AptosFramework
> INCLUDING DEPENDENCY AptosStdlib
> INCLUDING DEPENDENCY AxelarFramework
> INCLUDING DEPENDENCY MoveStdlib
> BUILDING hello_world
> error: property `map_borrow_mut_with_default` is not valid in this context
>    ┌─ /home/stephen/.move/https___github_com_aptos-labs_aptos-core_git_main/aptos-move/framework/aptos-framework/../aptos-stdlib/sources/table.spec.move:7:9
>    │
>  7 │ ╭         pragma intrinsic = map,
>  8 │ │             map_new = new,
>  9 │ │             map_destroy_empty = destroy,
> 10 │ │             map_has_key = contains,
>    · │
> 19 │ │             map_spec_del = spec_remove,
> 20 │ │             map_spec_has_key = spec_contains;
>    │ ╰─────────────────────────────────────────────^
> 
> error: property `map_borrow_mut_with_default` is not valid in this context
>    ┌─ /home/stephen/.move/https___github_com_aptos-labs_aptos-core_git_main/aptos-move/framework/aptos-framework/../aptos-stdlib/sources/table_with_length.spec.move:7:9
>    │
>  7 │ ╭         pragma intrinsic = map,
>  8 │ │             map_new = new,
>  9 │ │             map_destroy_empty = destroy_empty,
> 10 │ │             map_len = length,
>    · │
> 22 │ │             map_spec_len = spec_len,
> 23 │ │             map_spec_has_key = spec_contains;
>    │ ╰─────────────────────────────────────────────^
> 
> {
>   "Error": "Move compilation failed: extended checks failed"
> }
> Compiling, may take a little while to download git dependencies...
> UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
> INCLUDING DEPENDENCY AptosFramework
> INCLUDING DEPENDENCY AptosStdlib
> INCLUDING DEPENDENCY AxelarFramework
> INCLUDING DEPENDENCY MoveStdlib
> BUILDING token_linker
> error: property `map_borrow_mut_with_default` is not valid in this context
>    ┌─ /home/stephen/.move/https___github_com_aptos-labs_aptos-core_git_main/aptos-move/framework/aptos-framework/../aptos-stdlib/sources/table.spec.move:7:9
>    │
>  7 │ ╭         pragma intrinsic = map,
>  8 │ │             map_new = new,
>  9 │ │             map_destroy_empty = destroy,
> 10 │ │             map_has_key = contains,
>    · │
> 19 │ │             map_spec_del = spec_remove,
> 20 │ │             map_spec_has_key = spec_contains;
>    │ ╰─────────────────────────────────────────────^
> 
> error: property `map_borrow_mut_with_default` is not valid in this context
>    ┌─ /home/stephen/.move/https___github_com_aptos-labs_aptos-core_git_main/aptos-move/framework/aptos-framework/../aptos-stdlib/sources/table_with_length.spec.move:7:9
>    │
>  7 │ ╭         pragma intrinsic = map,
>  8 │ │             map_new = new,
>  9 │ │             map_destroy_empty = destroy_empty,
> 10 │ │             map_len = length,
>    · │
> 22 │ │             map_spec_len = spec_len,
> 23 │ │             map_spec_has_key = spec_contains;
>    │ ╰─────────────────────────────────────────────^
> 
> {
>   "Error": "Move compilation failed: extended checks failed"
> }
> ```

Fixed in the latest commit. 🙇‍♂️

**Review by StephenFluin — APPROVED** [review]
Wohoo! LGTM

**Pull request merged**